src/Chapter2.hs

```diff
@@ -841,7 +855,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate num l = if num < 0 then [] else subList (getStartIndex num (length l)) (getEndIndex num (length l)) (cycle l)
+                where
+                  getStartIndex num lsize = mod num lsize
+                  getEndIndex num lsize = (getStartIndex num lsize) + lsize - 1
```

Comment:
`num` is in scope from the `rotate` function argument, so you are actually shadowing that one. Instead, you could amend passing it to this function at all, and use it from the scope. The same applies to the `length l` 🙂